### PR TITLE
Add util to parse an NFTokenID

### PR DESCRIPTION
**Pull request by JST5000**
## High Level Overview of Change

Add a utility method to parse NFTokenIDs into a readable format.

### Context of Change

In response to #1897 

### Type of Change

- [X] New feature (non-breaking change which adds functionality)

## Test Plan

Added unit tests for this new feature.

### CODE REVIEW

**Review by mvadari — APPROVED** [review]

**Comment by JST5000** [comment]
@mvadari @ckniffen Sorry, could you re-review this? I realized my tests weren't actually being run because I named them `_test...` instead of `test` and found some issues with my decoding. (I was decoding to strings, when I needed to decode the number values straight to numbers. (`1024 -> 0x400` instead of being encoded digit by digit like `01000204`)

**Review by ckniffen — APPROVED** [review]

File: tests/unit/utils/test_parse_nftoken_id.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+from unittest import TestCase
+
+from xrpl.utils.parse_nftoken_id import parse_nftoken_id
+
+
+class TestParseNFTokenID(TestCase):
+    """Test parse_nftoken_id."""
+
+    def test_parse_nftoken_id_successful(self: TestParseNFTokenID) -> None:
+        nft_id = "000B0539C35B55AA096BA6D87A6E6C965A6534150DC56E5E12C5D09E0000000C"  # noqa:E501
```
Comment (mvadari) [inline tests/unit/utils/test_parse_nftoken_id.py:13]: This shouldn't be triggering E501 - it's only 84 characters long and the limit is set to 88.

File: xrpl/utils/parse_nftoken_id.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+    Returns:
+        A decoded nft TokenID with all information encoded within
+
+    # noqa:E501
+    """
+    expected_length = 64
+    print(len(nft_id))
+    if len(nft_id) != expected_length:
+        raise RuntimeError(
```
Comment (mvadari) [inline xrpl/utils/parse_nftoken_id.py:82]: This should be `XrplError`

**Review by khancode — APPROVED** [review]

File: xrpl/utils/parse_nftoken_id.py [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+    Raises:
+        XRPLException: when given an invalid Token ID as nft_id.
+
+    Returns:
+        A decoded nft TokenID with all information encoded within
+
+    # noqa:E501
+    """
+    expected_length = 64
+    print(len(nft_id))
```
Comment (mvadari) [inline xrpl/utils/parse_nftoken_id.py:81]: this should be removed

**Review by mvadari — APPROVED** [review]
lgtm other than the debug artifact

**Pull request merged**